src/app/Services/product.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core';
+import {Card, Product} from "../interfaces";
+import {BehaviorSubject, Observable} from "rxjs";
+import {products} from "../constants";
+
+@Injectable({
+    providedIn: 'root'
+})
+
+export class ProductService {
+
+    private cardListSubject = new BehaviorSubject<Card[]>([]);
```

Comment:
```suggestion
    private _cardListSubject = new BehaviorSubject<Card[]>([]);
```
Les variables private et protected sont préfixées de _